tests/torch/quantization/test_functions.py

```diff
@@ -200,51 +314,92 @@ def test_quantize_symmetric_forward(self, _seed, is_signed, is_weights, is_fp16,
+            else:
+                check_outputs_for_quantization_functions(test_value, ref_value, is_fp16,
+                                                         rtol=1e-2 if is_fp16 else 1e-3)
 
         def test_quantize_symmetric_backward(self, _seed, is_signed, is_weights, is_fp16, input_size, bits, use_cuda,
                                              scale_mode):
             if not torch.cuda.is_available() and use_cuda is True:
                 pytest.skip("Skipping CUDA test cases for CPU only setups")
             skip_if_half_on_cpu(is_fp16, use_cuda)
-            ref_input = generate_input(input_size)
+            if is_fp16:
+                np_dtype = np.float16
+            else:
+                np_dtype = np.float32
 
-            ref_scale = self.generate_scale(ref_input, scale_mode, is_weights)
+            fixed = None
+            if is_fp16:
+                # This is needed to make scale == 1 to prevent
+                # quant movement on forward pass in FP16 precision.
+                # In case scale != 1., not precice scale multiplication in FP16
+                # could lead to big deviations, so even if an input point
+                # lies in safe range (far from middles of quants) after a scaling
+                # it could end up in the middle of a quant. It happens mostly
+                # when target quant > 150 because in real life scenarious quantization range
+                # usualy less than 2 ** quantization bits,
+                # so input is small and scale is big, small FP16 input multiplies big fp16 scale,
+                # deviation is significant.
+                fixed = 2 ** (bits - 1) - 1
+            ref_scale = self.generate_scale(input_size, scale_mode, is_weights, fixed=fixed).astype(np_dtype)
+            test_scale = get_test_data([ref_scale], use_cuda, is_backward=True, is_fp16=is_fp16)
             level_low, level_high, levels = self.get_range_level(is_signed, bits)
-            test_input, test_scale = get_test_data([ref_input, ref_scale], use_cuda, is_backward=True,
-                                                   is_fp16=is_fp16)
 
             ref_scale = abs(ref_scale) + EPS
-            if is_fp16:
-                ref_input = ref_input.astype(np.float16)
-                ref_scale = ref_scale.astype(np.float16)
 
             ref_input_low = ref_scale * (level_low / level_high)
             ref_input_range = ref_scale - ref_input_low
 
+            # This is needed to prevent middle points in forward pass.
+            # Small deviation in computations could lead to completely different
+            # results: instead of quant a quant a + 1 will be returned.
+            # Different results in forward leads to high deviations on bacward pass.
+            # To prevent this, input values are put into [min_deviation, max_deviation]
+            # section of quant, so small deviation woun't change the quant on forward pass
```

Comment:
```suggestion
            # Different results in forward leads to high deviations on backward pass.
            # To prevent this, input values are put into [min_deviation, max_deviation]
            # section of quant, so small deviation won't change the quant on forward pass
```
really minor

Reply:
Thank you, fixed